### PR TITLE
Toggle rear camera in video call

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
PTAL @charlesmchen 

I'm waiting on the final button asset, but that should be a trivial drop in.

### CODE REVIEW

**Comment by charlesmchen-signal** [comment]
Screenshots?

File: Signal/src/ViewControllers/CallViewController.swift [hunk 1]
```diff
@@ -345,42 +348,60 @@ class CallViewController: OWSViewController, CallObserver, CallServiceObserver,
 
 //        textMessageButton = createButton(imageName:"message-active-wide",
 //                                                action:#selector(didPressTextMessage))
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/CallViewController.swift]: Remove?

File: Signal/src/ViewControllers/CallViewController.swift [hunk 2]
```diff
@@ -345,42 +348,60 @@ class CallViewController: OWSViewController, CallObserver, CallServiceObserver,
...
-        audioSourceButton = createButton(imageName: "audio-call-speaker-inactive",
+        audioSourceButton = createButton(image: #imageLiteral(resourceName: "audio-call-speaker-inactive"),
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/CallViewController.swift]: I'm not sure how I feel about these.

File: Signal/src/call/PeerConnectionClient.swift [hunk 3]
```diff
@@ -220,6 +224,21 @@ class PeerConnectionClient: NSObject, RTCPeerConnectionDelegate, RTCDataChannelD
         self.videoSender = videoSender
     }
 
+    public func setCameraSource(useBackCamera: Bool) {
+        guard let localVideoSource = self.localVideoSource else {
+            owsFail("\(logTag) in \(#function) localVideoSource was unexpectedly nil")
+            return
+        }
+
+        // certain devices, e.g. 16GB iPod touch don't have a back camera
```
Comment (charlesmchen-signal) [inline Signal/src/call/PeerConnectionClient.swift]: Should we add one to our test matrix?

File: Signal/src/ViewControllers/CallViewController.swift (hunk 2 above)
Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/CallViewController.swift]: It renders it in the IDE, which is cute, but not really that helpful. 

The main reason I like them is that they do a better job than logic like this:

```
 let image = UIImage(named: imageName)	
 assert(image != nil)
```

e.g. while developing you can be pretty sure you didn't mistype the name, and while running you know that it exists.

File: Signal/src/call/PeerConnectionClient.swift (hunk 3 above)
Comment (michaelkirk-signal) [reply Signal/src/call/PeerConnectionClient.swift]: Personally, I think this accounts for such a small user base that I'm not inclined to dedicate QA resources to verifying it.

**Comment by michaelkirk-signal** [comment]
> Screenshots?

Whoops - I took a screenshot, but never posted it.

after_video_2

**Pull request merged**